async_downloads/ws_consumers.py

```diff
@@ -0,0 +1,104 @@
+        {
+            "type": msg_type,
+            "data": {
+                "download": download,
+                "downloadKey": download_key,
+            },
+        },
+    )
+
+
+def ws_init_download(download_key):
+    _ws_group_send("init_download", download_key)
+
+
+def ws_update_download(download_key):
+    _ws_group_send("update_single_download", download_key)
+
+
+class DownloadsConsumer(AsyncWebsocketConsumer):
+    async def connect(self):
+        self.username = self.scope["user"].username
+        self.user = await database_sync_to_async(get_user_model().objects.get)(
+            username=self.username
+        )
+        self.user_group_name = f"{WS_CHANNEL_NAME}_{self.username}"
+        await self.channel_layer.group_add(self.user_group_name, self.channel_name)
+        await self.accept()
+
+    async def disconnect(self, close_code):
+        await self.channel_layer.group_discard(self.user_group_name, self.channel_name)
+
+    async def receive(self, text_data):
+        data = json.loads(text_data)
+        event = data["eventType"]
+        if event == "clearDownload":
+            await self.clear_download(data["data"]["filepath"])
+        if event == "initializeAllDownloads":
+            await self.initialize_all_downloads()
+
+    async def initialize_all_downloads(self):
+        from async_downloads.cache import get_collection_key
+
+        download_keys = cache.get(get_collection_key(self.user), [])
+        downloads = []
+        for download_key in download_keys:
+            dl = cache.get(download_key)
+            if not dl:
+                continue
+            if dl["complete"]:
+                dl["url"] = default_storage.url(dl["filepath"])
+            dl["timestamp"] = str(dl["timestamp"])
+            downloads.append({"download": dl, "downloadKey": download_key})
+        await self.send(
+            json.dumps(
+                {"eventType": settings.EVENT_TYPE_INIT_ALL_DOWNLOADS, "data": downloads}
+            )
+        )
+
+    async def remove_single_download(self, data):
+        await self.send(
+            json.dumps({"eventType": settings.EVENT_TYPE_REMOVE_DOWNLOAD, **data})
+        )
+
+    async def update_single_download(self, data):
+        await self.send(
+            json.dumps({"eventType": settings.EVENT_TYPE_UPDATE_DOWNLOAD, **data})
+        )
+
+    async def init_download(self, data):
+        await self.send(
+            json.dumps({"eventType": settings.EVENT_TYPE_INIT_DOWNLOAD, **data})
+        )
+
+    async def clear_download(self, filepath):
+        directory = os.path.split(filepath)[0]
+        download_key = os.path.split(directory)[1]
+        cache.delete(download_key)
+        default_storage.delete(filepath)
+        default_storage.delete(directory)
+        await self.channel_layer.group_send(
+            self.user_group_name,
+            {"type": "remove_single_download", "data": {"downloadKey": download_key}},
```

Comment:
```suggestion
            {"eventType": settings.EVENT_TYPE_REMOVE_DOWNLOAD, "data": {"downloadKey": download_key}},
```
Should this be like this?

Reply:
yes, because that is channels internal trigger so the function inside consumer with name 'remove_single_download' will be triggered for all of the sessions